### DISCUSSION TITLE
How to create a cross compilation file that uses CC, CXX etc. from the environment?

### DISCUSSION THREAD

**Discussion by criatura2** [mesonbuild/meson#14361] [General]
Hi, I want to cross compile on a build machine `x86_64-linux-gnu` to a host `aarch64-poky-linux`.  For that I want to create a meson `cross.ini` file that should have as content something similar to

```
[binaries]
c = 'CC' # How to pass this properly?
cpp = 'CXX'
...

[host_machine]
system = 'linux'
cpu_family = 'VAR'
```

However, I don't know how to actually pass the `CC`, `CXX` and other environment variables to `cross.ini`. Is there a simple way of doing it or do I have to invent something e.g. using `sed`?

Thanks

**Comment by taozuhong** [comment]
`meson env2mfile` output file should give some tips to implement the cross compilation.
